src/libs/actions/PersonalDetails.js

```diff
@@ -310,6 +310,54 @@ function openPersonalDetailsPage() {
     API.read('OpenPersonalDetailsPage');
 }
 
+/**
+ * Fetches public profile info about a given user
+ * The API will only return the accountID, displayName, and avatar for the user
+ * But the profile page will use other info (e.g. contact methods and pronouns) if they are already available in Onyx
+ * @param {Number} accountID
+ */
+function openPublicProfilePage(accountID) {
+    API.read(
+        'OpenPublicProfilePage',
+        {accountID},
+        {
+            optimisticData: [
+                {
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+                    value: {
+                        [accountID]: {
+                            isLoading: true,
+                        },
+                    },
+                },
+            ],
+            successData: [
+                {
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+                    value: {
+                        [accountID]: {
+                            isLoading: false,
+                        },
+                    },
+                },
+            ],
+            failureData: [
+                {
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+                    value: {
+                        [accountID]: {
+                            isLoading: false,
+                        },
+                    },
+                },
+            ],
+        },
+    );
```

Comment:
Suuuuper NAB, I find this format a bit clearer since it uses less lines:

```suggestion
    const optimisticData = [{
        onyxMethod: Onyx.METHOD.MERGE,
        key: ONYXKEYS.PERSONAL_DETAILS_LIST,
        value: {
            [accountID]: {
                isLoading: true,
            },
        },
    }];
    const successData = [{
        onyxMethod: Onyx.METHOD.MERGE,
        key: ONYXKEYS.PERSONAL_DETAILS_LIST,
        value: {
            [accountID]: {
                isLoading: false,
            },
        },
    }];
    const failureData = [{
        onyxMethod: Onyx.METHOD.MERGE,
        key: ONYXKEYS.PERSONAL_DETAILS_LIST,
        value: {
            [accountID]: {
                isLoading: false,
            },
        },
    }];
    API.read('OpenPublicProfilePage', {accountID}, {optimisticData, successData, failureData});
```